src/bindings.cpp

```diff
@@ -86,26 +134,59 @@ extern "C"
         return player.get()->isInited() ? 1 : 0;
     }
 
-    /// Load a new sound to be played once or multiple times later
+    /// Load a new sound to be played once or multiple times later.
     ///
-    /// [completeFileName] the complete file path
+    /// After loading the file, the [fileLoadedCallback] will call the 
+    /// Dart function defined with [setDartEventCallback] which gives back
+    /// the error and the new hash.
+    ///
+    /// [completeFileName] the complete file path.
     /// [loadIntoMem] if true Soloud::wav will be used which loads
     /// all audio data into memory. This will be useful when
     /// the audio is short, ie for game sounds, mainly used to prevent
     /// gaps or lags when starting a sound (less CPU, more memory allocated).
     /// If false, Soloud::wavStream will be used and the audio data is loaded
     /// from the given file when needed (more CPU, less memory allocated).
     /// See the [seek] note problem when using [loadIntoMem] = false
-    /// [hash] return hash of the sound
+    /// [hash] return the hash of the sound
     /// Returns [PlayerErrors.noError] if success
-    FFI_PLUGIN_EXPORT enum PlayerErrors loadFile(
+    FFI_PLUGIN_EXPORT void loadFile(
         char *completeFileName,
-        bool loadIntoMem,
+        bool loadIntoMem)
+    {
+        // this check is already been done in Dart
+        if (!player.get()->isInited())
+            return;
```

Comment:
Can we assert here instead? I'd rather this crash in development than to have some kind of bug live on.

Reply:
This function is called by FFI from Dart where, in `soloud.dart` we are always checking if the player is initilized:
```dart
if (!isInitialized) {
    throw const SoLoudNotInitializedException();
}
```
So this C function is never called if on Dart side the player is not initialized.

Here the check is a plus which make me feel better :).

Reply:
Yes, but if we ever change the Dart code and introduce a bug (i.e. calling `loadFile` before initialization), then the call fails silently (nothing will work when it should have, and there will be no indication there's a problem). For that reason, I'm a big proponent of "fail fast" - throw the error. Don't keep on trucking. I know the error throwing is redundant at this point, but it might not be in the future.

Reply:
Ok, I could assert every C function instead which does that check :)

Reply:
I'm afraid I don't know enough about C++ conventions here. In Dart, if I knew it's never going to be tested in debug mode, I'd just `throw StateError('Player has not been initialized');`.

Reply:
I can throw in C but I don't know how to catch it in Dart. This means that if C throws, the app crashes.
Sorry I'd leave it as is for now and I write down a note if you don't mind.

Reply:
I have searched and found [this issue](https://github.com/dart-lang/sdk/issues/45462#issuecomment-806438576). Is not possible (yet?) to catch from native C. The only way is to return some kind of error as some other relevant functions in this source do by returning `enum PlayerErrors`.

Reply:
Ok, I was happy if the C++ thing throws and crashes _without_ it being sent to the Dart code. Just [fail fast principle](https://en.wikipedia.org/wiki/Fail-fast_system). But maybe that means that the C++ side just crashes silently and the Dart developer never knows?

If that's the case, you may be right that the best thing to do is to print a warning onto the console.

```cpp
printf("WARNING (from SoLoud C++ binding code): the player has not yet been initialized. This is likely a bug in flutter_soloud. Please report the bug.");
```

Or something like that.